core/server/api_container/server/startosis_engine/recipe/extractor_test.go

```diff
@@ -0,0 +1,61 @@
+package recipe
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"go.starlark.net/starlark"
+	"testing"
+)
+
+var jsonObject = []byte(`
+{
+	"integer": 1,
+	"str": "a",
+	"list": [0.2, "b"],
+	"dict": {
+		"key": "value",
+		"another_key": -1
+	},
+	"bool": false
+}
+`)
+
+func TestExtractor_Success(t *testing.T) {
+	logrus.SetLevel(logrus.DebugLevel)
+	expectedDict := starlark.NewDict(2)
+	_ = expectedDict.SetKey(starlark.String("key"), starlark.String("value"))
+	_ = expectedDict.SetKey(starlark.String("another_key"), starlark.MakeInt(-1))
+	queryList := []string{".integer", ".str", ".list", ".dict.key", ".bool", ".dict", ".integer, .str"}
+	expectedList := []starlark.Comparable{
+		starlark.MakeInt(1),
+		starlark.String("a"),
+		starlark.NewList([]starlark.Value{starlark.Float(0.2), starlark.String("b")}),
+		starlark.String("value"),
+		starlark.Bool(false),
+		expectedDict,
+		starlark.NewList([]starlark.Value{starlark.MakeInt(1), starlark.String("a")}),
+	}
+	for i := range queryList {
+		result, err := Extractor(queryList[i], jsonObject)
+		assert.Nil(t, err)
+		assert.Equal(t, expectedList[i], result)
+	}
+}
+
+func TestExtractor_FailureQuery(t *testing.T) {
```

Comment:
not a jq wizard, but if i'm not mistaken, here the query itself is correct, but it doesn't match any field, right? Can we add a test with an invalid query (syntax error) and make sure it throws?

Reply:
Done! Good catch